File: app/src/main/java/com/examle/curexchange/domain/ExchangeInteractor.java
package com.examle.curexchange.domain;

import com.examle.curexchange.data.repository.exchange.ExchangeRepository;

import javax.inject.Inject;

import io.reactivex.Flowable;
import io.reactivex.Observable;

public class ExchangeInteractor {

    private ExchangeRepository exchangeRepository;

    @Inject
    public ExchangeInteractor(ExchangeRepository exchangeRepository) {
        this.exchangeRepository = exchangeRepository;
    }

    public Flowable<Float> getResult(String firstName, String secondName, int value) {
        return exchangeRepository.getResult(firstName, secondName, value);
    }
}

File: app/src/main/java/com/examle/curexchange/domain/ExchangeInteractor.kt
package com.examle.curexchange.domain

import com.examle.curexchange.data.repository.exchange.ExchangeRepository
import io.reactivex.Flowable
import javax.inject.Inject

class ExchangeInteractor @Inject constructor(private val exchangeRepository: ExchangeRepository) {

    fun getResult(firstCurrencyName: String,
                  secondCurrencyName: String,
                  value: Int): Flowable<Float> =
            exchangeRepository.getResult(firstCurrencyName, secondCurrencyName, value)
}